src/Compilers/Shared/CoreClrAnalyzerAssemblyLoader.cs

```diff
@@ -8,19 +8,29 @@ namespace Microsoft.CodeAnalysis
 {
     internal sealed class CoreClrAnalyzerAssemblyLoader : AnalyzerAssemblyLoader
     {
-        private readonly AssemblyLoadContext _loadContext;
+        private AssemblyLoadContext _loadContext;
 
         public CoreClrAnalyzerAssemblyLoader()
         {
-            _loadContext = AssemblyLoadContext.GetLoadContext(typeof(CoreClrAnalyzerAssemblyLoader).GetTypeInfo().Assembly);
+        }
 
-            _loadContext.Resolving += (context, name) =>
+        protected override Assembly LoadFromPathImpl(string fullPath)
+        {
+            // Initialize the _loadContext in lazy fashion
+            if (_loadContext == null)
```

Comment:
Because .NET Native doesn't support GetLoadContext and it was in the .ctor which is being called. LoadFromPathImpl is not called in the .NET Native build, so I opted for the lazy initialization.